lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -0,0 +1,581 @@
+    this.lowBalanceNoticeDays = config.lowBalanceNoticeInDays || 28;
+  }
+
+  /**
+   * generates a private key in WIF format
+   * @param network Which bitcoin network to generate this key for
+   */
+  public static generatePrivateKey (network: 'mainnet' | 'livenet' | 'testnet' | undefined): string {
+    let bitcoreNetwork: Networks.Network | undefined;
+    switch (network) {
+      case 'mainnet':
+        bitcoreNetwork = Networks.mainnet;
+        break;
+      case 'livenet':
+        bitcoreNetwork = Networks.livenet;
+        break;
+      case 'testnet':
+        bitcoreNetwork = Networks.testnet;
+        break;
+    }
+    return new PrivateKey(undefined, bitcoreNetwork).toWIF();
+  }
+
+  /**
+   * Initializes the Bitcoin processor
+   */
+  public async initialize () {
+    console.debug('Initializing TransactionStore');
+    await this.transactionStore.initialize();
+    const lastKnownTransaction = await this.transactionStore.getLastTransaction();
+    let syncedTo: {height: number, hash: string };
+    if (lastKnownTransaction) {
+      console.info(`Last known block ${lastKnownTransaction.transactionTime} (${lastKnownTransaction.transactionTimeHash})`);
+      syncedTo = await this.processTransactions(lastKnownTransaction.transactionTime, lastKnownTransaction.transactionTimeHash);
+    } else {
+      syncedTo = await this.processTransactions();
+    }
+    this.lastBlockHash = syncedTo.hash;
+    this.lastBlockHeight = syncedTo.height;
+    this.periodicPoll();
+  }
+
+  /**
+   * Gets the latest logical blockchain time.
+   * @param hash time blockchain time hash
+   * @returns the current or associated blockchain time and blockchain hash
+   */
+  public async time (hash?: string): Promise<IBlockchainTime> {
+    console.info(`Getting time ${hash ? 'since' + hash : ''}`);
+    let request: any;
+    if (hash) {
+      request = {
+        method: 'getblock',
+        params: [
+          hash, // hash of the block
+          true, // block details
+          false // transaction details
+        ]
+      };
+    } else {
+      const tip = await this.getTip();
+      request = {
+        method: 'getblockbyheight',
+        params: [
+          tip,  // height of the block
+          true, // block details
+          false // transaction details
+        ]
+      };
+    }
+    const response = await this.rpcCall(request);
+    return {
+      hash: response.hash,
+      time: response.height
+    };
+  }
+
+  /**
+   * Fetches Sidetree transactions in chronological order from since or genesis.
+   * @param since A transaction number
+   * @param hash The associated transaction time hash
+   * @returns Transactions since that blocktime
+   */
+  public async transactions (since?: number, hash?: string): Promise<{
+    moreTransactions: boolean,
+    transactions: ITransaction[]
+  }> {
+    if (since && !hash) {
+      throw new SidetreeError(httpStatus.BAD_REQUEST);

```

Comment:
It is my intention to not have direct "http" reference in the lib, consider using the internal `ResponseStatus` enum (which I acknowledge that I want to make it string enums after our discussion yesterday.)

Reply:
For sure!
I was more or less just switching between the existing implementation, the implementation doc, and how Koa would need the error to be formatted to return correctly. As a true library, it should just be an enum so I'll make that change.